debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorTask.java

```diff
@@ -205,6 +216,89 @@ public synchronized void start(Map<String, String> props) {
         }
     }
 
+    private static MySqlTaskContext createAndStartTaskContext(Configuration config,
+                                                              Filters filters) {
+        MySqlTaskContext taskContext = new MySqlTaskContext(config, filters);
+        taskContext.start();
+        return taskContext;
+    }
+
+    /**
+     * @return true if new tables appear to have been added to the config, and false otherwise.
+     */
+    private boolean newTablesInConfig() {
+
+        final String elementSep = "/s*,/s*";
+
+        // take in two stringified lists, and return the parsed set of all elements exclusive to the first.
+        BiFunction<String, String, Set<String>> exclude = (String a, String b) -> {
+            if (a == null || a.isEmpty()) {
+                return new HashSet<>();
+            } else if (b == null || b.isEmpty()) {
+                return new HashSet<>(Arrays.asList(a.split(elementSep)));
+            }
+            Set<String> ASet = new HashSet<>(Arrays.asList(a.split(elementSep)));
+            Set<String> BSet = new HashSet<>(Arrays.asList(b.split(elementSep)));
+            ASet.removeAll(BSet);
+            return ASet;
+        };
+
+        // return true if stringified list a contains elements exclusive to it, that are not in stringified element list b.
+        BiFunction<String, String, Boolean> exclusiveElements = (String a, String b) -> !exclude.apply(a, b).isEmpty();
+
+        final SourceInfo sourceInfo = taskContext.source();
+        final Configuration config = taskContext.config;
+        if (!sourceInfo.hasFilterInfo()) {
+            // if there was previously no filter info, then we either can't evaluate if there are new tables,
+            // or there aren't any new tables because we previously used no filter.
+            return false;
+        }
+        // otherwise, we have filter info
+        // if either whitelist has been added to, then we may have new tables
+
+        if (exclusiveElements.apply(config.getString(MySqlConnectorConfig.DATABASE_WHITELIST), sourceInfo.getDatabaseWhitelist())) {
+            return true;
+        }
+        if (exclusiveElements.apply(config.getString(MySqlConnectorConfig.TABLE_WHITELIST), sourceInfo.getTableWhitelist())) {
+            return true;
+        }
+        // if either blacklist has been removed from, then we may have new tables
+        if (exclusiveElements.apply(sourceInfo.getDatabaseBlacklist(), config.getString(MySqlConnectorConfig.DATABASE_BLACKLIST))) {
+            return true;
+        }
+        if (exclusiveElements.apply(sourceInfo.getTableBlacklist(), config.getString(MySqlConnectorConfig.TABLE_BLACKLIST))) {
+            return true;
+        }
+        // otherwise, false.
+        return false;
```

Comment:
Probably I'm missing something here, but why is it that functions are used here that heavily here? Couldn't be simply done like that:

```
Set<String> previousDatabaseBlacklistPatterns = Collect.asSet(sourceInfo.getDatabaseBlacklist().split(elementSep));
Set<String> currentDatabaseBlacklistPatterns = Collect.asSet(config.getString(MySqlConnectorConfig.DATABASE_BLACKLIST).split(elementSep));
...

boolean allEqual = previousDatabaseBlacklistPatterns.equals(currentDatabaseBlacklistPatterns) &&
    previousDatabaseWhitelistPatterns.equals(currentDatabaseWhitelistPatterns) &&
    previousTableBlacklistPatterns.equals(currentTableBlacklistPatterns) &&
    previousTableWhitelistPatterns.equals(currentTableWhitelistPatterns);

return !allEqual;

Reply:
I'd still be curious about the answer to that one :)